### PR TITLE
Update ParameterType and BuiltInParameterGroup because of RevitAPI ch…

### PR DESCRIPTION
**Pull request by ZiyunShang**
…anged.

### Purpose

1. Update ParameterType, BuiltInParameterGroup and GetSpecTypeId(). Some API will be removed from Revit in future, so we also will remove the Obsolete Nodes in future.
2. Add ForgeType to store Revit ForgeTypeId.
3. Update related tests.

### Declarations

Check these if you believe they are true

- [ ] The code base is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] Snapshot of UI changes, if any.

### Reviewers

@wangyangshi @ShengxiZhang 

### FYIs

@QilongTang @mjkkirschner @Amoursol

### CODE REVIEW

File: src/Libraries/RevitNodesUI/RevitTypes.cs [hunk 1]
```diff
@@ -122,17 +122,35 @@ public RevisionNumberType(IEnumerable<PortModel> inPorts, IEnumerable<PortModel>
     [NodeDescription("ParameterTypeSelectorDescription", typeof(DSRevitNodesUI.Properties.Resources))]
     [IsDesignScriptCompatible]
     [OutPortTypes("string")]
+    [IsVisibleInDynamoLibrary(false)]
+    [Obsolete("This node may remove in future version. Use SpecTypes instead.")]
     public class ParameterType : CustomGenericEnumerationDropDown
```
Comment (ZiyunShang) [inline src/Libraries/RevitNodesUI/RevitTypes.cs:127]: Hi @QilongTang , I wanted to set this Node to Obsolete, but found that **NodeObsolete** cannot be used outside of methods and properties.
I just make this node - 'Select Parameter Type' invisible in Dynamo Library, and update its NodeDescription.
Do you have any suggestions here?

File: src/Libraries/RevitNodes/Elements/Parameter.cs [hunk 2]
```diff
@@ -262,6 +283,82 @@ public static void CreateSharedParameter(string parameterName, string groupName,
...
+        {
+            CreateSharedParameter(parameterName, groupName, specType, group, instance, null);
+        }
+
+        /// <summary>
+        /// Create a new Shared Parameter in the current Revit document
+        /// </summary>
+        /// <param name="parameterName">Name</param>
+        /// <param name="groupName">Group of the parameter for shared parameters</param>
+        /// <param name="specType">Spec Type</param>
```
Comment (ShengxiZhang) [inline src/Libraries/RevitNodes/Elements/Parameter.cs:304]: Keep consistent, as well as line 454

File: src/Libraries/RevitNodes/Elements/Parameter.cs [hunk 3]
```diff
@@ -347,6 +446,76 @@ public static void CreateProjectParameter(string parameterName, string groupName
...
         }
 
+        /// <summary>
+        /// Create a new Project Parameter in this current Revit document for all applicable categories
+        /// </summary>
+        /// <param name="parameterName">Name</param>
+        /// <param name="groupName">Group of the parameter for shared parameters</param>
+        /// <param name="specType">Parameter Type</param>
+        /// <param name="group">Parameter Group</param>
+        /// <param name="instance">Is instance parameter, otherwise it's a type parameter</param>
```
Comment (ShengxiZhang) [inline src/Libraries/RevitNodes/Elements/Parameter.cs:456]: True if it's an instance parameter...

File: src/Libraries/RevitNodes/Elements/Parameter.cs [hunk 4]
```diff
@@ -347,6 +446,76 @@ public static void CreateProjectParameter(string parameterName, string groupName
...
+
+        /// <summary>
+        /// Create a new Project Parameter in this current Revit document
+        /// </summary>
+        /// <param name="parameterName">Name</param>
+        /// <param name="groupName">Group of the parameter for shared parameters</param>
+        /// <param name="specType">Spec Type</param>
+        /// <param name="group">Group Type</param>
+        /// <param name="instance">Is instance parameter, otherwise it's a type parameter</param>
+        /// <param name="categoryList">List of categories this parameter applies to. If no category is supplied, all possible categories are selected</param>
```
Comment (ShengxiZhang) [inline src/Libraries/RevitNodes/Elements/Parameter.cs:470]: A list..

File: src/Libraries/RevitNodes/Properties/Resources.resx [hunk 5]
```diff
@@ -573,4 +573,31 @@
...
+  </data>
+  <data name="GroupObsolete" xml:space="preserve">
+    <value>This property may remove in future version. Use GroupType instead.</value>
+  </data>
+  <data name="ParameterTypeObsolete" xml:space="preserve">
+    <value>This property may remove in future version. Use SpecType instead.</value>
+  </data>
+  <data name="UnitTypeObsolete" xml:space="preserve">
+    <value>This property may remove in future version. Use SpecType instead.</value>
+  </data>
```
Comment (ShengxiZhang) [inline src/Libraries/RevitNodes/Properties/Resources.resx:602]: `may be removed...`
for all the messages above.

**Review by ShengxiZhang — APPROVED** [review]

**Review by wangyangshi — APPROVED** [review]

**Pull request merged**